test/runtests.jl

```diff
@@ -343,6 +343,7 @@ include("metrics.jl")
 include("sampling.jl")
 include("seeding.jl")
 include("site_selection.jl")
+include("constrain_reef_clusters.jl")
```

Comment:
How about a more generic name like "spatial_clustering.jl"?

Reply:
Sure, sounds good. I made variable names in the function based on `reef_` but I can change this to be more generic also if that's best?

Reply:
Inside the function is fine, the point is that the file is more generic so we can add other tests relevant to spatial clustering later.